hazelcast/src/test/java/com/hazelcast/config/YamlRootAdditionalPropertiesTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.client.config.YamlClientConfigBuilderTest;
+import com.hazelcast.internal.config.SchemaViolationConfigurationException;
+import com.hazelcast.test.OverridePropertyRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static com.hazelcast.config.YamlConfigBuilderTest.buildConfig;
+import static com.hazelcast.test.OverridePropertyRule.clear;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+public class YamlRootAdditionalPropertiesTest {
+
+    @Rule
+    public ExpectedException expExc = ExpectedException.none();
+
+    @Rule
+    public OverridePropertyRule indentationCheckEnabled = clear("hazelcast.yaml.config.indentation.check.enabled");
```

Comment:
Minor: You can retrieve the name from the property here as well.

Reply:
I would prefer keeping it this way here (in case someone modifies the property name by accident, the test will catch it).